### PR TITLE
ENH: special: improve accuracy of `zetac` for negative arguments

### PR DESCRIPTION
**Pull request by person142**
Combine the reflection formula for zeta with the Lanczos approximation
for Gamma to prevent overflow. Closes gh-8371.

### CODE REVIEW

**Comment by person142** [comment]
Travis failure is a timeout; note that the full test suite job finished so the new mpmath test did run.

**Review by pv — APPROVED** [review]
lgtm

File: scipy/special/cephes/zetac.c [hunk 1]
```diff
@@ -162,37 +163,67 @@ static double S[5] = {
     7.43853965136767874343E4,
 };
 
+static double TAYLOR0[10] = {
```
[the remark this answers is not in the document]
Comment (person142) [reply scipy/special/cephes/zetac.c]: Sure. We actually still need to `const` the `coeff` argument in `polevl`, which currently has the signature `polevl(double x, double coeff[], int N)`.

**Comment by person142** [comment]
I also switched to using `fmod` for the `sin` argument reduction; not doing it before was just a temporary lapse in sanity ;-)

**Pull request merged**